Makefile

```diff
@@ -88,13 +88,13 @@ build-llvm:    $(llvm_kompiled)
 build-haskell: $(haskell_kompiled)
 
 $(llvm_kompiled): $(llvm_defn)
-	$(K_BIN)/kompile --backend llvm                                           \
+	kompile --backend llvm                                                    \
```

Comment:
Is this relevant to this PR, or just cleanup? I see that in the beginning of the Makefile we are exporting a new path with the `K_BIN` on top anyway, so I guess it isn't necessary at all.

Reply:
We need this, because in the Docker image K is not installed at `deps/k/...`, but at `/usr/bin`.

Reply:
So is the `export PATH` in the Makefile a potential problem? If we were to build the K dependency (even if we don't do that now I guess that could change later) that would introduce a new `kompile` binary in the path and we would silently switch to using that one.

It seems the "correct" way to to this would be to have the Jenkins-file set the `K_BIN` variable when calling `make build`.

Reply:
Well, the Makefile should work in all these scenarios:

-   User sets `K_RELEASE`, and so manually picks which version of K to select from (this is for the Haskell backend team, which often uses out-of-tree builds of K).
-   User doesn't set `K_RELEASE`, but builds the submodule, we should prefer the submodule build of K then.
-    User doesn't set `K_RELEASE`, and doesn't build the submodule, but has a global install of K available, we should then use the global install of K.

Reply:
This isn't quite what happens, but I can correct this PR (and others) once the K PR gets in.